src/components/LocationAutocomplete/LocationAutocomplete.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import { FaMapMarkerAlt } from 'react-icons/fa';
+import { Autosuggest, Alert, ListItem, MarkedText } from '../../index';
+import useDebounce from '../../hooks/useDebounce';
+import POWERED_BY_GOOGLE_ON_WHITE from '../../images/powered_by_google_on_white.png';
+import styles from './LocationAutocomplete.scss';
+
+const { elem } = bem({
+    name: 'LocationAutocomplete',
+    classnames: styles,
+});
+
+const DEBOUNCE_DELAY = 350;
+const ACCEPTABLE_API_STATUSES = ['OK', 'NOT_FOUND', 'ZERO_RESULTS'];
+
+const LocationAutocomplete = props => {
+    const [inputValue, setInputValue] = React.useState('');
+    const [suggestionsList, setSuggestionsList] = React.useState(null);
+    const [isLoading, setIsLoading] = React.useState(false);
+
+    const {
+        onSelectionChange,
+        inputPlaceholder,
+        noSuggestionsPlaceholder,
+        country,
+        placeTypes,
+        showCountryInSuggestions,
+        onError,
+        ...rest
+    } = props;
+
+    // Suggestion functions
+    const resetSuggestionsList = () => setSuggestionsList(null);
+    const suggestionToString = suggestion => (suggestion ? suggestion.description : '');
+
+    const debouncedInputValue = useDebounce(inputValue, DEBOUNCE_DELAY);
+
+    React.useEffect(() => {
+        if (debouncedInputValue) {
+            const service = new window.google.maps.places.AutocompleteService();
+
+            service.getPlacePredictions(
+                {
+                    input: debouncedInputValue,
+                    types: placeTypes,
+                    componentRestrictions: { country },
+                },
+                (predictions, status) => {
+                    if (ACCEPTABLE_API_STATUSES.includes(status)) {
+                        setSuggestionsList(predictions);
```

Comment:
**Consider slow connection scenario:**

1. Connection is not super fast
2. User types `Ams`, it triggers request `A`
3. While request `A` is still in progress, user deletes `Ams` and types `Lon`, that triggers request `B`.
4. At this point request `A` is completed and user sees suggestions for `Ams`,  having `Lon` in the input. That's inconsistent state of UI (in this scenario it's temporary).
5. Later request `B` is completed and only now user sees suggestions for `Lon`

**Worse scenario:**

1. Connection is not very stable
2. User types `Ams`, it triggers request `A`
3. While request `A` is still in progress, user deletes `Ams` and types `Lon`, that triggers request `B`.
4. At this point request `B` is completed and user sees suggestions for `Lon` for a moment
5. Moment later request `A` is completed and user sees suggestions for `Ams` having `Lon` in the input. That's also inconsistent state of UI (and in this scenario it's already a bug).